modules/ipc/include/hephaestus/ipc/zenoh/session.h

```diff
@@ -18,9 +19,40 @@ namespace heph::ipc::zenoh {
 enum class Mode : uint8_t { PEER = 0, CLIENT, ROUTER };
 enum class Protocol : uint8_t { ANY = 0, UDP, TCP };
 
+/// There are a lot of options to configure a zenoh session,
+/// See  for more information https://zenoh.io/docs/manual/configuration/#configuration-files
+struct ZenohConfig {
+  ZenohConfig() {
+    zconfig.insert_json5(Z_CONFIG_ADD_TIMESTAMP_KEY, "true");
+  }
+  explicit ZenohConfig(std::string const& path) : zconfig(::zenoh::Config::from_file(path)) {
+  }
+
+  ~ZenohConfig() noexcept = default;
+  ZenohConfig(const ZenohConfig&) = delete;
+  auto operator=(const ZenohConfig&) -> ZenohConfig& = delete;
+  ZenohConfig(ZenohConfig&&) noexcept = default;
+  auto operator=(ZenohConfig&&) noexcept -> ZenohConfig& = default;
```

Comment:
Is there a downside in allowing for copy constructor?

Reply:
Yes, the zenoh configuration option is only movable, so it is implicitly deleted already. This is just for clarity.